Warnings

~~~
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘unsigned int deal2lkit::IDAInterface<VEC>::start_ode(VEC&, VEC&, unsigned int) [with VEC = dealii::BlockVector<double>]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:532:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:374:18: warning: variable ‘frac’ set but not used [-Wunused-but-set-variable]
           double frac = 0;
                  ^
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘unsigned int deal2lkit::IDAInterface<VEC>::start_ode(VEC&, VEC&, unsigned int) [with VEC = dealii::TrilinosWrappers::MPI::Vector]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:537:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:374:18: warning: variable ‘frac’ set but not used [-Wunused-but-set-variable]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘unsigned int deal2lkit::IDAInterface<VEC>::start_ode(VEC&, VEC&, unsigned int) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:538:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:374:18: warning: variable ‘frac’ set but not used [-Wunused-but-set-variable]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_lsetup(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::BlockVector<double>; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:495:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::BlockVector<double>]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:532:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp1’ [-Wunused-parameter]
 int t_dae_lsetup(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp2’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp3’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_solve(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::BlockVector<double>; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:496:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::BlockVector<double>]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:532:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:173:5: warning: unused parameter ‘weight’ [-Wunused-parameter]
 int t_dae_solve(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_lsetup(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::TrilinosWrappers::MPI::Vector; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:495:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::TrilinosWrappers::MPI::Vector]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:537:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp1’ [-Wunused-parameter]
 int t_dae_lsetup(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp2’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp3’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_solve(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::TrilinosWrappers::MPI::Vector; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:496:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::TrilinosWrappers::MPI::Vector]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:537:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:173:5: warning: unused parameter ‘weight’ [-Wunused-parameter]
 int t_dae_solve(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_lsetup(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:495:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:538:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp1’ [-Wunused-parameter]
 int t_dae_lsetup(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp2’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp3’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_solve(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:496:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:538:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:173:5: warning: unused parameter ‘weight’ [-Wunused-parameter]
 int t_dae_solve(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘unsigned int deal2lkit::IDAInterface<VEC>::start_ode(VEC&, VEC&, unsigned int) [with VEC = dealii::BlockVector<double>]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:532:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:374:18: warning: variable ‘frac’ set but not used [-Wunused-but-set-variable]
           double frac = 0;
                  ^
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘unsigned int deal2lkit::IDAInterface<VEC>::start_ode(VEC&, VEC&, unsigned int) [with VEC = dealii::TrilinosWrappers::MPI::Vector]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:537:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:374:18: warning: variable ‘frac’ set but not used [-Wunused-but-set-variable]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘unsigned int deal2lkit::IDAInterface<VEC>::start_ode(VEC&, VEC&, unsigned int) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:538:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:374:18: warning: variable ‘frac’ set but not used [-Wunused-but-set-variable]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_lsetup(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::BlockVector<double>; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:495:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::BlockVector<double>]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:532:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp1’ [-Wunused-parameter]
 int t_dae_lsetup(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp2’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp3’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_solve(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::BlockVector<double>; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:496:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::BlockVector<double>]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:532:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:173:5: warning: unused parameter ‘weight’ [-Wunused-parameter]
 int t_dae_solve(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_lsetup(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::TrilinosWrappers::MPI::Vector; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:495:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::TrilinosWrappers::MPI::Vector]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:537:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp1’ [-Wunused-parameter]
 int t_dae_lsetup(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp2’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp3’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_solve(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::TrilinosWrappers::MPI::Vector; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:496:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::TrilinosWrappers::MPI::Vector]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:537:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:173:5: warning: unused parameter ‘weight’ [-Wunused-parameter]
 int t_dae_solve(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_lsetup(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:495:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:538:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp1’ [-Wunused-parameter]
 int t_dae_lsetup(IDAMem IDA_mem,
     ^
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp2’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc:144:5: warning: unused parameter ‘tmp3’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/ida_interface.cc: In instantiation of ‘int deal2lkit::t_dae_solve(IDAMem, N_Vector, N_Vector, N_Vector, N_Vector, N_Vector) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector; IDAMem = IDAMemRec*; N_Vector = _generic_N_Vector*]’:
/home/heltai/src/deal2lkit/source/ida_interface.cc:496:23:   required from ‘void deal2lkit::IDAInterface<VEC>::reset_ode(double, VEC&, VEC&, double, unsigned int, bool) [with VEC = dealii::TrilinosWrappers::MPI::BlockVector]’
/home/heltai/src/deal2lkit/source/ida_interface.cc:538:27:   required from here
/home/heltai/src/deal2lkit/source/ida_interface.cc:173:5: warning: unused parameter ‘weight’ [-Wunused-parameter]
 int t_dae_solve(IDAMem IDA_mem,
~~~

--- a/source/ida_interface.cc
+++ b/source/ida_interface.cc
@@ -149,7 +149,9 @@ int t_dae_lsetup(IDAMem IDA_mem,
                  N_Vector tmp2,
                  N_Vector tmp3)
 {
-
+  (void) tmp1;
+  (void) tmp2;
+  (void) tmp3;
   SundialsInterface<VEC> &solver = *static_cast<SundialsInterface<VEC> *>(IDA_mem->ida_user_data);
 
   shared_ptr<VEC> src_yy = solver.create_new_vector();
@@ -177,6 +179,7 @@ int t_dae_solve(IDAMem IDA_mem,
                 N_Vector yp,
                 N_Vector resp)
 {
+  (void) weight;
   SundialsInterface<VEC> &solver = *static_cast<SundialsInterface<VEC> *>(IDA_mem->ida_user_data);
 
   shared_ptr<VEC> src_yy = solver.create_new_vector();
@@ -371,10 +374,10 @@ unsigned int IDAInterface<VEC>::start_ode(VEC &solution,
 
       while ( reset == true )
         {
-          double frac = 0;
+          // double frac = 0;
           int k = 0;
           IDAGetLastOrder(ida_mem, &k);
-          frac = std::pow((double)k,2.);
+          // frac = std::pow((double)k,2.);
           reset_ode(t, solution, solution_dot,
                     h/2.0, max_steps, false);
           reset = solver.solver_should_restart(t, step_number, h, solution, solution_dot);

